plugins/unifying/fu-unifying-bootloader.c

```diff
@@ -75,14 +75,45 @@ fu_unifying_bootloader_parse_requests (FuUnifyingBootloader *self, GBytes *fw, G
 		}
 		payload->addr = ((guint16) fu_unifying_buffer_read_uint8 (tmp + 0x03)) << 8;
 		payload->addr |= fu_unifying_buffer_read_uint8 (tmp + 0x05);
+		payload->cmd = FU_UNIFYING_BOOTLOADER_CMD_WRITE_RAM_BUFFER;
 
 		rec_type = fu_unifying_buffer_read_uint8 (tmp + 0x07);
 
-		/* record type of 0xFD indicates signature data */
-		if (rec_type == 0xFD) {
-			payload->cmd = FU_UNIFYING_BOOTLOADER_CMD_WRITE_SIGNATURE;
-		} else {
-			payload->cmd = FU_UNIFYING_BOOTLOADER_CMD_WRITE_RAM_BUFFER;
+		switch (rec_type) {
+			case 0x00: /* data */
+				break;
+			case 0x01: /* EOF */
+				goto exit;
```

Comment:
I don't like the `goto` -- can we do a g_set_error() or a break with some kind of boolean flag?

Reply:
Sure.